File: practice23/2.cpp


//https://www.nowcoder.com/practice/8705437354604a7cb9ba7bf959e42de6?tpId=98&tqId=32855&tPage=2&rp=2&ru=/ta/2019test&qru=/ta/2019test/question-ranking
#include<iostream>
#include<algorithm>
using namespace std;

int main()
{
	int n, Max = -999999, sum = 0;
	while (cin >> n)
	{
		if (cin.get() == '\n')
		{
			sum = max(sum + n, n);
			Max = max(sum, Max);
			break;
		}
		sum = max(sum + n, n);
		Max = max(sum, Max);
	}
	if (Max > 0)
	{
		cout << Max << endl;
	}
	else
	{
		cout << 0 << endl;
	}

	system("pause");
	return 0;
}

File: practice34/2.java

//https://www.nowcoder.com/practice/e08819cfdeb34985a8de9c4e6562e724?tpId=46&tqId=29030&tPage=1&rp=1&ru=/ta/leetcode&qru=/ta/leetcode/question-ranking
class TreeNode {
    int val;
    TreeNode left;
    TreeNode right;

    TreeNode(int x) {
        val = x;
    }
}

class Solution {
    public int run(TreeNode root) {
        if (root == null) {
            return 0;
        }
        int l = run(root.left);
        int r = run(root.right);
        if (l == 0 || r == 0) {
            return l + r + 1;
        }
        return Math.min(r, l) + 1;
    }
}